util/process_wrapper/output.rs

```diff
@@ -0,0 +1,27 @@
+use std::io::{self, prelude::*};
+use std::thread;
+
+pub(crate) fn process_output<F>(
+    read_end: Box<dyn Read + Send>,
+    write_end: Box<dyn Write + Send>,
+    mut process_line: F,
+) -> thread::JoinHandle<io::Result<()>>
+where
+    F: FnMut(String) -> Option<String> + Send + 'static,
+{
+    thread::spawn(move || {
```

Comment:
Is it necessary to perform this work concurrently? I've had frustrating experiences with actions actions not playing nice with others by occupying a ton of resources on the host.